homeassistant/components/homematicip_cloud/alarm_control_panel.py

```diff
@@ -73,3 +85,104 @@ def state(self) -> str:
     async def async_alarm_arm_away(self, code=None):
         """Send arm away command."""
         await self._home.set_security_zones_activation(True, True)
+
+
+class HomematicipAlarmControlPanel(AlarmControlPanel):
+    """Representation of an alarm control panel."""
+
+    def __init__(self, home: AsyncHome, security_zones) -> None:
+        """Initialize the alarm control panel."""
+        self._home = home
+        self.alarm_state = STATE_ALARM_DISARMED
+
+        for security_zone in security_zones:
+            if security_zone.label == 'INTERNAL':
+                self._internal_alarm_zone = security_zone
+            else:
+                self._external_alarm_zone = security_zone
+
+    @property
+    def state(self) -> str:
+        """Return the state of the device."""
+        activation_state = self._home.get_security_zones_activation()
+        # check arm_away
+        if activation_state == (True, True):
+            if self._internal_alarm_zone_state or \
+                    self._external_alarm_zone_state:
+                return STATE_ALARM_TRIGGERED
+            return STATE_ALARM_ARMED_AWAY
+        # check arm_home
+        if activation_state == (False, True):
+            if self._external_alarm_zone_state:
+                return STATE_ALARM_TRIGGERED
+            return STATE_ALARM_ARMED_HOME
+
+        return STATE_ALARM_DISARMED
+
+    @property
+    def _internal_alarm_zone_state(self) -> bool:
+        return _get_zone_alarm_state(self._internal_alarm_zone)
+
+    @property
+    def _external_alarm_zone_state(self) -> bool:
+        """Return the state of the device."""
+        return _get_zone_alarm_state(self._external_alarm_zone)
+
+    async def async_alarm_disarm(self, code=None):
+        """Send disarm command."""
+        await self._home.set_security_zones_activation(False, False)
+
+    async def async_alarm_arm_home(self, code=None):
+        """Send arm home command."""
+        await self._home.set_security_zones_activation(False, True)
+
+    async def async_alarm_arm_away(self, code=None):
+        """Send arm away command."""
+        await self._home.set_security_zones_activation(True, True)
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        self._internal_alarm_zone.on_update(self._device_changed)
+        self._external_alarm_zone.on_update(self._device_changed)
+
+    def _device_changed(self, *args, **kwargs):
```

Comment:
For clarity, decorate this with `@callback`, or change the name to `_async_device_changed`.

Reply:
I also rename the same in device.py